NETCORE/src/Shared/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -316,6 +316,13 @@ private static void AddCommonInitializers(IServiceCollection services)
 
         private static void AddCommonTelemetryModules(IServiceCollection services)
         {
+            // Previously users were encouraged to manually add the DiagnosticsTelemetryModule.
+            // They could have added this either as an INSTANCE or as a TYPE. 
+            if (!services.Any(o => typeof(IHeartbeatPropertyManager).IsAssignableFrom(o.ImplementationType ?? o.ImplementationInstance.GetType())))
```

Comment:
Logically, adding `DiagnosticsTelemetryModule` should check if `DiagnosticsTelemetryModule` is already present or not. Why checking IHeartbeatPropertyManager?

Please have a comment explaining this.

Reply:
I fixed this and moved the logic to an extension method (with tests) to clean it up.